src/extra/others/fragment/lv_fragment.h

```diff
@@ -0,0 +1,232 @@
+/**
+ * @file lv_fragment.h
+ *
+ */
+
+#ifndef lv_fragment_H
+#define lv_fragment_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "../../../lv_conf_internal.h"
+
+#if LV_USE_FRAGMENT
+
+#include "../../../core/lv_obj.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+typedef struct _lv_fragment_manager_t lv_fragment_manager_t;
+
+typedef struct lv_fragment_t lv_fragment_t;
+typedef struct lv_fragment_class_t lv_fragment_class_t;
+
+/**
+ * Opaque pointer for internal state management
+ */
+typedef struct manager_stack_t manager_stack_t;
+
+struct lv_fragment_t {
+    /**
+     * Class of this fragment
+     */
+    const lv_fragment_class_t * cls;
+    /**
+     * Manager this fragment instance belongs to
+     */
+    lv_fragment_manager_t * manager;
+    /**
+     * lv_obj returned by create_obj_cb
+     */
+    lv_obj_t * obj;
+
+    /**
+     * Private internal states
+     */
+    manager_stack_t * priv_item;
```

Comment:
It seems `manager_stack_t` and the fragment is tightly coupled. What is the point of `manager_stack_t` as pointer instead of having its fields directly in `lv_fragment_t`?  I'd help to unify many static functions: one works on `item` and other on `fragment`.

Reply:
There are too many internal states in this manager_stack_t. If we expose them all, I'm afraid users will be confused when they see too many fields at once.

Reply:
In other parts of LVGL we provide API function to access struct members, so if something needs to be hidden simply there is no API function for it.

But if you'd like to keep it separate I think it'd be better to add `manager_stack_t` as a field of `lv_fragment_t` (e.g. `lv_fragment_priv_t priv_data`) and not a pointer in order to really manage them together.

Reply:
This internal state only available when fragment is managed, so I'll stay with pointer. But yes, the struct can be made non-opaque. It's just not meant to be modify by user, but fragment manager only.